variables.tf

```diff
@@ -390,3 +390,15 @@ variable "bootstrap_extra_args" {
   default     = ""
   description = "Passed to the bootstrap.sh script to enable --kublet-extra-args or --use-max-pods."
 }
+
+variable "aws_iam_instance_profile" {
+  type        = "string"
+  default     = ""
+  description = "Use this to provide already existed instance profile that will be used in autoscaling group for EKS workers, if empty will create instance profile for you."
+}
+
+variable "use_provided_aws_iam_instance_profile" {
```

Comment:
i think we don't need this var.
let's add this:

locals {
  use_existing_instance_profile = "${var.aws_iam_instance_profile_name  != "" ? true : false}"
}

and then in all `counts` use:

count = "${var.enabled == "true" && local.use_existing_instance_profile == false ? 1 : 0}"

iam_instance_profile_name = "${local.use_existing_instance_profile == false ? join("", aws_iam_instance_profile.default.*.name) : var.aws_iam_instance_profile_name}"
  
aws_iam_role_arn = "${local.use_existing_instance_profile == true ?  join("", data.aws_iam_instance_profile.default.*.role_arn) : join("", aws_iam_role.default.*.arn)}"

Reply:
thanks its much better to use computed variable rather than add additional, looks like its working